packages/driver/src/cypress.ts

```diff
@@ -182,7 +183,11 @@ class $Cypress {
 
     // set domainName but allow us to turn
     // off this feature in testing
-    if (domainName && config.testingType === 'e2e') {
+    const shouldInjectDocumentDomain = cors.shouldInjectDocumentDomain(window.location.origin, {
```

Comment:
Isn't this false when `experimentalUseDefaultDocumentDomain` is true? I'm hesitant to pull this module into the driver code. Most of the cors code is expected to run in node not the browser right?

Reply:
I think it would be better to update the logic in the data-context/server to no sent the `domainName` name if this should not be injected by default into a new Cypress instance.

Reply:
> Isn't this false when experimentalUseDefaultDocumentDomain is true?
the other way around. if `experimentalUseDefaultDocumentDomain` is truthy and fits the glob patterns, then `shouldInjectDocumentDomain` should return false.

The cors code is already referenced in the driver quite a bit. I think the expectation is that is should run in the browser or node, but not sure if that was the original intention 😅 .

Reply:
@AtofStryker is it the driver or just the driver tests? I wasn't finding the dependency used in the shipped code when i searched for it, but I must be missing it

Reply:
I'm hesitating to modify this in the data-context and server for the manipulated domainName value because it isn't exactly clear why that value is being manipulated since it would happen in the driver. The config flag in the `configure` method is localized, which I think makes the impact/expected change clearer.

Reply:
noticed I had a bug in the cors package around the `shouldInjectDocumentDomain` function where it wasn't returning false on glob matches. going to add some unit tests here

Reply:
fixed the bug in 861de35229